android/src/com/mapswithme/maps/location/LocationHelper.java

```diff
@@ -338,6 +340,24 @@ public static void onLocationUpdated(@NonNull Location location)
                           location.getBearing());
   }
 
+  /**
+   * Obtains last known location regardless of "My position" button state.
+   * @return {@code null} on failure.
+   */
+  public @Nullable Location getLastKnownLocation()
+  {
+    if (mLastLocation != null)
+      return mLastLocation;
+
+    AndroidNativeProvider provider = new AndroidNativeProvider();
+    List<String> providers = provider.getFilteredProviders();
+
+    if (providers.isEmpty())
+      return null;
+
+    return provider.findBestNotExpiredLocation(providers, LocationUtils.LOCATION_EXPIRATION_TIME_MILLIS_LONG);
```

Comment:
Смена региона у пользователя между запусками наших карт - это не самая частая операция. Текущий таймаут в несколько минут уже точно ничего не поменяет: тогда лучше вообще не мержить этот реквест.

> On Apr 1, 2016, at 12:43, Dmitry Yunitsky <notifications@github.com> wrote:
> 
> In android/src/com/mapswithme/maps/location/LocationHelper.java <https://github.com/mapsme/omim/pull/2672#discussion_r58183802>:
> 
> > +  /**
> > +   * Obtains last known location regardless of "My position" button state.
> > +   * @return {@code null} on failure.
> > +   */
> > +  public @Nullable Location getLastKnownLocation()
> > +  {
> > +    if (mLastLocation != null)
> > +      return mLastLocation;
> > +
> > +    AndroidNativeProvider provider = new AndroidNativeProvider();
> > +    List<String> providers = provider.getFilteredProviders();
> > +
> > +    if (providers.isEmpty())
> > +      return null;
> > +
> > +    return provider.findBestNotExpiredLocation(providers, LocationUtils.LOCATION_EXPIRATION_TIME_MILLIS_LONG);
> А мне кажется это достаточно важно, и совсем отказавшись от проверки устаревания позиции может получиться неприятно для пользователя. Если мы удаляем все страны, и качаем какой-то соседний регион, не тот, где действительно находится юзер - он остаётся вообще без карты текущей. И может даже не заметить этого, пока не вернётся на карту и не увидит, что нифига не работает.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2672/files/cc92bbc4b790b789ab9d08f4d3d2825564b6054a#r58183802>